plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadata.java

```diff
@@ -260,7 +262,11 @@ public Optional<ProjectionApplicationResult<ConnectorTableHandle>> applyProjecti
                 .collect(toImmutableList());
 
         if (handle.getColumns().isPresent()) {
-            Set<JdbcColumnHandle> newColumnSet = ImmutableSet.copyOf(newColumns);
+            Set<JdbcColumnHandle> newColumnSet = newColumns.stream()
+                    // It may happen fresh table handle comes with a columns prepared already.
+                    // In such case it may happen that applyProjection may want to add UPDATE_ROW_ID id, which is created later during the planning.
+                    .filter(column -> !column.getColumnName().equals(UPDATE_ROW_ID))
```

Comment:
UPDATE_ROW_ID could be different for different connectors. So this will not work for some, which have `getDeleteRowIdColumnHandle` overriden. 
Please consider 
```
            Set<JdbcColumnHandle> tableSyntheticColumnSet = ImmutableSet.<JdbcColumnHandle>builder()
                    .addAll(tableColumnSet)
                    .add((JdbcColumnHandle) getDeleteRowIdColumnHandle(session, table))
                    .add((JdbcColumnHandle) getMergeRowIdColumnHandle(session, table))
                    .add((JdbcColumnHandle) getUpdateRowIdColumnHandle(session, table, ImmutableList.of())) // not sure about ImmutableList.of()
                    .build();

            verify(tableSyntheticColumnSet.containsAll(newColumnSet), "applyProjection called with columns %s and some are not available in existing query: %s", newColumnSet, tableSyntheticColumnSet);
```

Reply:
and actually these will fail:
```
          .add((JdbcColumnHandle) getMergeRowIdColumnHandle(session, table))
          .add((JdbcColumnHandle) getUpdateRowIdColumnHandle(session, table, ImmutableList.of())) // not sure about ImmutableList.of()
```
because they `throw new TrinoException(NOT_SUPPORTED`

Reply:
So may be add a new public method `List<String> getSyntacticColumnNames()` 
and `.filter(column -> !getSyntacticColumnNames().contains(column.getColumnName())`

Reply:
Let's not overcomplicate things